hack/run-ccm-migration-e2e-test-in-kind.sh

```diff
@@ -0,0 +1,313 @@
+    debugLog: true
+  featureGates:
+    TunnelingExposeStrategy: {}
+  ui:
+    replicas: 0
+  # Dex integration
+  auth:
+    #tokenIssuer: "http://dex.oauth:5556/dex"
+    #issuerRedirectURL: "http://localhost:8000"
+    tokenIssuer: "https://127.0.0.1.nip.io/dex"
+    serviceAccountKey: "$(generate_secret)"
+EOF
+
+HELM_VALUES_FILE="${TMPDIR}/values.yaml"
+cat << EOF > ${HELM_VALUES_FILE}
+dex:
+  replicas: 0
+kubermaticOperator:
+  image:
+    repository: "quay.io/kubermatic/kubermatic${REPOSUFFIX}"
+    tag: "${TAG}"
+EOF
+
+# install dependencies and Kubermatic Operator into cluster
+./_build/kubermatic-installer deploy \
+  --storageclass copy-default \
+  --config "${KUBERMATIC_CONFIG}" \
+  --kubeconfig "${KUBECONFIG}" \
+  --helm-values "${HELM_VALUES_FILE}" \
+  --helm-binary "${HELM_BINARY}"
+
+# TODO: The installer should wait for everything to finish reconciling.
+#echodate "Waiting for Kubermatic Operator to deploy Master components..."
+# sleep a bit to prevent us from checking the Deployments too early, before
+# the operator had time to reconcile
+sleep 5
+retry 10 check_all_deployments_ready kubermatic
+
+echodate "Finished installing Kubermatic"
+
+echodate "Installing Seed..."
+SEED_MANIFEST="${TMPDIR}/seed.yaml"
+
+SEED_KUBECONFIG="$(cat ${KUBECONFIG} | sed 's/127.0.0.1.*/kubernetes.default.svc.cluster.local./' | base64 -w0)"
+
+cat << EOF > ${SEED_MANIFEST}
+kind: Secret
+apiVersion: v1
+metadata:
+  name: "${SEED_NAME}-kubeconfig"
+  namespace: kubermatic
+data:
+  kubeconfig: "${SEED_KUBECONFIG}"
+
+---
+kind: Seed
+apiVersion: kubermatic.k8s.io/v1
+metadata:
+  name: "${SEED_NAME}"
+  namespace: kubermatic
+  labels:
+    worker-name: ""
+spec:
+  country: Germany
+  location: Hamburg
+  kubeconfig:
+    name: "${SEED_NAME}-kubeconfig"
+    namespace: kubermatic
+    fieldPath: kubeconfig
+  datacenters:
+    syseleven-dbl1:
+      country: DE
+      location: Syseleven - dbl1
+      node: {}
+      spec:
+        openstack:
+          auth_url: https://api.cbk.cloud.syseleven.net:5000/v3
+          availability_zone: dbl1
+          dns_servers:
+          - 37.123.105.116
+          - 37.123.105.117
+          enabled_flavors: null
+          enforce_floating_ip: true
+          ignore_volume_az: false
+          images:
+            centos: kubermatic-e2e-centos
+            coreos: kubermatic-e2e-coreos
+            flatcar: flatcar
+            ubuntu: kubermatic-e2e-ubuntu
+          manage_security_groups: null
+          node_size_requirements:
+            minimum_memory: 0
+            minimum_vcpus: 0
+          region: dbl
+          trust_device_path: null
+          use_octavia: null
+  expose_strategy: Tunneling
+EOF
+
+retry 3 kubectl apply -f $SEED_MANIFEST
+echodate "Finished installing Seed"
+
+sleep 5
+echodate "Waiting for Kubermatic Operator to deploy Seed components..."
+retry 8 check_all_deployments_ready kubermatic
+echodate "Kubermatic Seed is ready."
+
+apply_api_server_nodeport &
```

Comment:
why is this a function if you call it only once? I mean, other manifests a being inlined but not this one.

Reply:
You are right, I'll remove the function